### PR TITLE
Update hugging_stanza.py to use `HfApi` instead of `Repository`

### PR DESCRIPTION
**Pull request by Wauplin**
Ping @AngledLuffa @osanseviero, following up our recent discussion.

This PR updates the script to take advantage of the "HTTP-only" methods. You shouldn't encounter git-related issues anymore (here is [an explanation of git vs http approaches](https://huggingface.co/docs/huggingface_hub/concepts/git_vs_http)). A noticeable change with this script is that now the generated README file is added to the local `src` folder instead of the `dst_repo_path` before (since there is no local duplication anymore). If that's too much of a trouble, we can find a solution to update the script.

And please test this script before merging! I am not aware of the workflow here so only tried my best to have something that should run. Please let me know how it goes :)

### CODE REVIEW

**Comment by AngledLuffa** [comment]
Thank you!  I will give it a try

File: hugging_stanza.py [hunk 1]
```diff
@@ -82,52 +66,32 @@ def push_to_hub():
...
-        if repo.tag_exists(tag, remote=repo_url):
-            repo.delete_tag(tag, remote=repo_url)
-        repo.add_tag(tag_name=tag, message=f"Adding new version of models {tag}", remote=repo_url)
-        print(f"Added a tag for the new models: {tag}")
 
+        # Update model card in it
+        (src / "README.md).write_text(get_model_card(model))
+
+        # Upload model + model card
+        api.upload_folder(repo_id=repo_id, folder_path=src, version=f"Add model {args.version}")
```
Comment (AngledLuffa) [inline hugging_stanza.py:83]: not `version`?  will have to check the API

File: hugging_stanza.py [hunk 2]
```diff
@@ -82,52 +66,32 @@ def push_to_hub():
...
-        repo.push_to_hub(commit_message=f"Add model {args.version}")
-
-        tag = "v" + args.version
-        if repo.tag_exists(tag, remote=repo_url):
-            repo.delete_tag(tag, remote=repo_url)
-        repo.add_tag(tag_name=tag, message=f"Adding new version of models {tag}", remote=repo_url)
-        print(f"Added a tag for the new models: {tag}")
 
+        # Update model card in it
+        (src / "README.md).write_text(get_model_card(model))
```
Comment (AngledLuffa) [inline hugging_stanza.py:80]: this line explains why my emacs syntax coloring wasn't working :P

File: hugging_stanza.py (hunk 1 above)
Comment (AngledLuffa) [reply hugging_stanza.py:83]: should this just still be `commit_message`?

Comment (AngledLuffa) [reply hugging_stanza.py:83]: is there a convenient way to push everything to a subdirectory?  our old layout was, for example, to have a models directory under the root directory, such as here:

https://huggingface.co/stanfordnlp/stanza-es/tree/main
models, .gitattributes, README.md

all the subdirectories are now under the main directory:

https://huggingface.co/stanfordnlp/stanza-mr/tree/main
backward_charlm, depparse, etc etc, default.zip, .gitattributes, README.md

probably the `path_in_repo` flag?  although then there will need to be two separate pushes, one for the models and one for the README.  alternatively we could either change the code that arranges the models into their current layout, which is probably better, or we could change the code that downloads the models, and i'm kinda too lazy to do that.  but i believe rearranging the models when building up the model distribution is not going to be very difficult

Comment (Wauplin) [reply hugging_stanza.py:83]: Oops yes it must be! `version` is not even a valid arg :see_no_evil: 

```suggestion
        api.upload_folder(repo_id=repo_id, folder_path=src, commit_message=f"Add model {args.version}")
```

Comment (Wauplin) [reply hugging_stanza.py:83]: > probably the path_in_repo flag? although then there will need to be two separate pushes, one for the models and one for the README. alternatively we could either change the code that arranges the models into their current layout, which is probably better, or we could change the code that downloads the models, and i'm kinda too lazy to do that. but i believe rearranging the models when building up the model distribution is not going to be very difficult

Yes it is `path_in_repo` that you should use!
If you need 2 commits or to rearrange the dist structure locally, that's no problem (either way is good IMO). You're taking care of it IIUC?

Comment (AngledLuffa) [reply hugging_stanza.py:83]: Yes, am mostly done!  Just need to verify that the new layout is happy - it takes a while to rearrange all the models, and there's never been a need for a model rearranging script that only does one language for testing purposes, so it just takes a little while

File: hugging_stanza.py [hunk 3]
```diff
@@ -82,52 +66,32 @@ def push_to_hub():
...
-
-        repo = Repository(repo_local_path, clone_from=repo_url)
-        # checkout "main" so that we know we are tracking files correctly
-        repo.git_checkout("main")
-        if not repo.is_repo_clean():
-            print(f"Repo {repo_local_path} is currently not clean.  Unwilling to proceed...")
-            break
-        repo.git_pull(rebase=True)
-
-        # Make sure jar files are tracked with LFS
```
Comment (AngledLuffa) [inline hugging_stanza.py:102]: I would wonder a bit if missing this block for new repos will cause it to not use lfs for .zip or .pt files.  My memory of when we first set this up was that it did not put these kinds of files under lfs by default.

**Comment by AngledLuffa** [comment]
I can see that, excellent.

Unfortunately, if I run the test of creating a new repo using the script, the script is unable to push to it.  I ran this test by copying the `stanza-mr` repo to `stanza-mrzzz`, and although it created a new empty repo at https://huggingface.co/stanfordnlp/stanza-mrzzz the script does not successfully push to it

```
Processing languages: ['mrzzz']
Processing mrzzz
Traceback (most recent call last):
  File "/u/nlp/anaconda/main/anaconda3/envs/stanza-1.2/lib/python3.7/site-packages/huggingface_hub/utils/_errors.py", line 259, in hf_raise_for_status
    response.raise_for_status()
  File "/u/nlp/anaconda/main/anaconda3/envs/stanza-1.2/lib/python3.7/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://huggingface.co/api/models/stanfordnlp/stanza-mrzzz/commit/main

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "huggingface-models/hugging_stanza.py", line 103, in <module>
    push_to_hub()
  File "huggingface-models/hugging_stanza.py", line 88, in push_to_hub
    api.upload_folder(repo_id=repo_id, folder_path=src, commit_message=f"Add model {args.version}")
  File "/u/nlp/anaconda/main/anaconda3/envs/stanza-1.2/lib/python3.7/site-packages/huggingface_hub/utils/_validators.py", line 120, in _inner_fn
    return fn(*args, **kwargs)
  File "/u/nlp/anaconda/main/anaconda3/envs/stanza-1.2/lib/python3.7/site-packages/huggingface_hub/hf_api.py", line 3242, in upload_folder
    parent_commit=parent_commit,
  File "/u/nlp/anaconda/main/anaconda3/envs/stanza-1.2/lib/python3.7/site-packages/huggingface_hub/utils/_validators.py", line 120, in _inner_fn
    return fn(*args, **kwargs)
  File "/u/nlp/anaconda/main/anaconda3/envs/stanza-1.2/lib/python3.7/site-packages/huggingface_hub/hf_api.py", line 2548, in create_commit
    hf_raise_for_status(commit_resp, endpoint_name="commit")
  File "/u/nlp/anaconda/main/anaconda3/envs/stanza-1.2/lib/python3.7/site-packages/huggingface_hub/utils/_errors.py", line 297, in hf_raise_for_status
    raise BadRequestError(message, response=response) from e
huggingface_hub.utils._errors.BadRequestError: 

Bad request for commit endpoint:
Your commit has failed. Please contact us. Request ID: Root=1-646d08c2-4e647f5e15a78e2c4e785a01
```

**Comment by AngledLuffa** [comment]
ps. that was done with a slightly updated version of the change in the `web_interface` branch of this repo

File: hugging_stanza.py (hunk 3 above)
Comment (AngledLuffa) [reply hugging_stanza.py:102]: ... creating new repos now includes .zip and .pt ... thank you for the update!

**Comment by Wauplin** [comment]
As explained in slack, the error comes from a invalid README file due to unknown language `mrzzz` in the metadata section. We created an internal issue to return a more explicit message from the server when this happens. In the meantime I think you can consider you script as working since it should work with all the existing models you have! :)

**Pull request merged**